### PR TITLE
#171923516 setup testing environment 

### PR DESCRIPTION
**Pull request by Habinezajanvier**
#### What does this PR do?
* Setting testing environment
#### Description of Task to be completed?
* Setting *jest* for testing
* Installing *supertest* for request testing
#### How should this be manually tested?
* Clone the repo and checkout `ch-setup-testing-environment-#171923516`
* run `npm install` to install dependencies
* fill `.env` with correct credentials
* Set your environment to testing by running
```javascript
//For MacOs & Git Bash
$export NODE_ENV="testing"  
```
```javascript
//For windiw Powershell
$env:NODE_ENV="testing"
```
```javascript
//For Window cmd
set NODE_ENV=testing
```
* run `npm test`
> you should see a pass of 3 test
#### Any background context you want to provide?
* If you want to make a test fail, you may change any of the test suit in /tests/controller.test.js
#### What are the relevant pivotal tracker stories?
#171923516

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,21 +4,32 @@
   "description": "application to register what you have lost and what you found to help each other",
   "main": "index.js",
   "dependencies": {
-    "express": "^4.17.1",
+    "@babel/register": "^7.9.0",
```
Comment (YvesIraguha) [inline package.json]: This package is a dev-dependency.

File: package.json [hunk 2]
```diff
@@ -4,21 +4,32 @@
...
     "dotenv": "^8.2.0",
-    "mongoose": "^5.9.5"
+    "express": "^4.17.1",
+    "mongoose": "^5.9.5",
+    "morgan": "^1.10.0"
```
Comment (YvesIraguha) [inline package.json:9]: This package is a devDepency

File: src/controllers/controller.js [hunk 3]
```diff
@@ -1,141 +1,31 @@
...
+    const document = await Lost.findOne({name: req.body.name});
+    if(document) return res.status(400).json({msg: 'document exist'});
+
+    const lost = new Lost({
+        name: req.body.name,
+        number: req.body.number
+    });
+    try {
+        const savedDoc  = await lost.save();
+        res.json({msg: "successfully saved"})
```
Comment (YvesIraguha) [inline src/controllers/controller.js]: Actually the record created should also be returned.

Comment (YvesIraguha) [reply src/controllers/controller.js]: In addition, when I create other records it keeps responding document exist, not sure if that is the intended behavior.

File: src/controllers/controller.js [hunk 4]
```diff
@@ -1,141 +1,31 @@
...
+router.get('/lost', async (req, res)=>{
+    try {
+        const findDoc = await Lost.find();
+        res.json(findDoc);
+    } catch (error) {
+        res.json({msg: error.message});
+    }
+})
+
+module.exports = router;
```
Comment (YvesIraguha) [inline src/controllers/controller.js]: Add a line at the end of each file

**Review by YvesIraguha — COMMENTED** [review]
Also, the tests fail, when I run them.

File: .vscode/settings.json [hunk 5]
```diff
@@ -0,0 +1,2 @@
+{
```
Comment (YvesIraguha) [inline .vscode/settings.json]: Remove this folder, please. Ignore it.

File: tests/controller.test.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
+const request = require('supertest');
+const Lost = require('../models/model');
+
```
Comment (YvesIraguha) [inline tests/controller.test.js:3]: Can you use `es6` please. `import` versus `require()`

**Review by YvesIraguha — CHANGES_REQUESTED** [review]

**Review by SimonImanigirimpuhwe — COMMENTED** [review]
I tried to run the test and unfortunately it failed. There were the total 3 fail.

**Review by YvesIraguha — APPROVED** [review]

**Pull request merged**